tests/Model/IndexInfoTest.php

```diff
@@ -139,4 +139,19 @@ public function testOffsetUnsetCannotBeCalled()
         $this->expectExceptionMessage('MongoDB\Model\IndexInfo is immutable');
         unset($info['v']);
     }
+
+    public function testGetters()
+    {
+        $info = new IndexInfo([
+            'v' => 1,
+            'key' => ['x' => 1],
+            'name' => 'x_1',
+            'ns' => 'foo.bar',
+        ]);
+
+        $this->assertFalse($info->is2dSphere());
+        $this->assertFalse($info->isGeoHaystack());
+        $this->assertFalse($info->isText());
+        $this->assertFalse($info->isBackground());
+    }
```

Comment:
Using the example specs I shared in https://github.com/mongodb/mongo-php-library/pull/521#discussion_r180154102, can you create tests based on the `testTtlIndex()` (and friends) for truthy return values? I think you can then also just add `$this->assertFalse()` for the new "is" methods to existing tests (see: `$this->assertFalse($info->isSparse());` in `testTtlIndex()`.

Since you already created functional tests, I think it's worth keeping those since we didn't have functional tests for creating all of these index types previously.